opencog/attention/experimental/HebbianCreationAgent.cc

```diff
@@ -0,0 +1,173 @@
+        return;
+
+    HandleSeq notAttentionalFocus;
+    int afb = as->get_attentional_focus_boundary();
+    as->get_handles_by_AV(back_inserter(notAttentionalFocus),0,afb);
+
+   // Retrieve the atoms in the AttentionalFocus
+    HandleSeq attentionalFocus;
+    as->get_handle_set_in_attentional_focus(back_inserter(attentionalFocus));
+
+    // Exclude the source atom
+    attentionalFocus.erase(std::remove(attentionalFocus.begin(),
+                                       attentionalFocus.end(),
+                                       source), attentionalFocus.end());
+
+    if (source == Handle::UNDEFINED)
+        return;
+
+    // Get the neighboring atoms, where the connecting edge
+    // is an AsymmetricHebbianLink in either direction
+    HandleSeq existingAsSource =
+            get_target_neighbors(source, ASYMMETRIC_HEBBIAN_LINK);
+    HandleSeq existingAsTarget =
+            get_source_neighbors(source, ASYMMETRIC_HEBBIAN_LINK);
+
+    // Get the set differences between the AttentionalFocus
+    // and the sets of existing sources and targets
+    std::sort(existingAsSource.begin(), existingAsSource.end());
+    std::sort(existingAsTarget.begin(), existingAsTarget.end());
+    std::sort(attentionalFocus.begin(), attentionalFocus.end());
+    HandleSeq needToBeSource = set_difference(attentionalFocus,
+                                              existingAsSource);
+    HandleSeq needToBeTarget = set_difference(attentionalFocus,
+                                              existingAsTarget);
+
+
+    int count = 0;
+
+    // Resulting in the sets of nodes that require
+    // a new AsymmetricHebbianLink in either direction
+    for (Handle atom : needToBeSource) {
+        if (atom == Handle::UNDEFINED)
+            continue;
+        addHebbian(atom,source);
+    }
+
+    for (Handle atom : needToBeTarget) {
+        if (atom == Handle::UNDEFINED)
+            continue;
+        addHebbian(source,atom);
+        count++;
+    }
+
+    std::default_random_engine generator;
+    std::uniform_int_distribution<int> distribution(0,notAttentionalFocus.size()-1);
+
+    //How many links outside the AF should be created
+    int farLinks = round(count / localToFarLinks);
+
+    Handle link = Handle::UNDEFINED;
+    Handle target = Handle::UNDEFINED;
+
+    //Pick a random target and create the link if it doesn't exist already
+    for (int i = 0; i < farLinks; i++) {
+        target = notAttentionalFocus[distribution(generator)];
+        link = as->get_handle(ASYMMETRIC_HEBBIAN_LINK, source, target);
+        if (link == Handle::UNDEFINED)
+            addHebbian(source,target);
+    }
+
+    //Check the ammount of HebbianLinks the Atom has
+    IncomingSet iset = source->getIncomingSet();
+    iset.erase(remove_if(iset.begin(),iset.end(),
+                            [](LinkPtr lp) -> bool {
+                            return !(classserver().isA(lp->getType(), HEBBIAN_LINK));
+                            }),iset.end());
```

Comment:
It would probably be faster to use `getIncomingSetByType`, instead of getting all of them, and then erasing.